docs/source/developer_tutorial.ipynb

```diff
@@ -0,0 +1,253 @@
+    "diff = DiffusionAnalyzer.from_xdatcar(xd, **params)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4331a507",
+   "metadata": {},
+   "source": [
+    "[`DiffusionAnalyzer`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py#L20) is a subclass of [`Analyzer`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/analyzer.py#L20) and so upon its initialization `Analyzer` is initialized first, `DiffusionAnalyzer` inherits from the `Analyzer`.\n",
+    "\n",
+    "```python\n",
+    "\n",
+    "class DiffusionAnalyzer(Analyzer):\n",
+    "\n",
+    "```\n",
+    "\n",
+    "\n",
+    "The `from_` class method of the `DiffusionAnalyzer` calls the Analyzer class `_from_` method via a super(), i.e. within [`from_universe`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py#L131)\n",
+    "\n",
+    "```python\n",
+    "\n",
+    "p = super()._from_universe(trajectory, specie, time_step, step_skip, dtype, dt, dimension, distance_unit,\n",
+    "                            specie_indices, masses, progress)\n",
+    "\n",
+    "```\n",
+    "\n",
+    "Within the `_from_` methods the respective parser is called for each frame in the trajectory, similarly to above there are 3 parser classes : \n",
+    "\n",
+    "\n",
+    "* [`MDAnalysis_parser`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/analyzer.py#L109) - For MDAnalysis Universe\n",
+    "* [`PymatgenParser`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/analyzer.py#L74) - For Pymatgen Structures\n",
+    "* [`ASE_parser`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/analyzer.py) - For ASE objects\n",
+    "\n",
+    "\n",
+    "Each respective parser class is a subclass of the full [`Parser`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/parser.py#L34) class. \n",
+    "\n",
+    "```python \n",
+    "\n",
+    "class PymatgenParser(Parser):\n",
+    "\n",
+    "```\n",
+    "\n",
+    "**The parser indexes the correct molecules and particle coordinates for a timestep.**\n",
+    "\n",
+    "**The Analyzer calls the parser for all the chosen time steps and calculates displacements.**\n",
+    "\n",
+    "**The DiffusionAnalyzer then calculates the mean squared displacements (MSD) and variances for each time interval.**\n",
+    "\n",
+    "A flowchart of the classes and inheritance can be seen below\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "525f47c4",
+   "metadata": {},
+   "source": [
+    "![My Diagram](./diffusion_analyzer_flowchart.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "43c158f2",
+   "metadata": {},
+   "source": [
+    "The main output of the `from_` methods is then [`.msd_da`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py#L83) which is a [Scipp](https://scipp.github.io/) DataArray containing the MSD, n_samples, time intervals as well as standard deviations, it can be accessed via"
```

Comment:
I would rephrase this as the object that results from the `from_*` method includes the `.msd_da` attribute, which is...